src/main/java/parsers/Parser.java

```
@@ -0,0 +1,117 @@
+package parsers;
+
+import exception.NicholasException;
+import tasks.TaskList;
+import tasks.ToDoTask;
+import tasks.DeadlineTask;
+import tasks.EventTask;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Parses user inputs into instructions
+ * <p>
+ * This class handles the inputs of users by processing and breaking the input down
+ * for the rest of the chatbot application.
+ */
+public class Parser {
+    private static final String LIST = "list";
+    private static final String MARK = "mark";
+    private static final String UNMARK = "unmark";
+    private static final String TODO = "todo";
+    private static final String DEADLINE = "deadline";
+    private static final String EVENT = "event";
+    private static final String DELETE = "delete";
+    private static final String FIND = "find";
+
+    /* Parse user input */
+    public void parseCommand(String input, TaskList tasks) throws NicholasException {
+
+        String command = input.split(" ")[0];
+
+        switch (command) {
+        case LIST -> {
+            tasks.getList();
+        }
+        case MARK -> {
+            int idx = prepareIndex(input);
+            tasks.markTaskAsDone(idx);
+        }
+        case UNMARK -> {
+            int idx = prepareIndex(input);
+            tasks.markTaskAsUndone(idx);
+        }
+        case TODO -> {
+            tasks.addItem(prepareToDo(input));
+        }
+        case DEADLINE -> {
+            try {
+                tasks.addItem(prepareDeadlineTask(input));
+            } catch (ArrayIndexOutOfBoundsException e) {
+                System.out.println("Please enter a deadline. e.g. /by Sunday");
```

Perhaps you could pass in your UI to parse to let your UI handle message printing? Or maybe you could just throw an exception instead like your default case.